crates/shared/src/utils.rs

```diff
@@ -116,8 +118,11 @@ impl<Types: NodeType, ApiVer: StaticVersionType> EventServiceStream<Types, ApiVe
     > {
         let client = Client::<hotshot_events_service::events::Error, ApiVer>::new(url.clone());
 
-        if !(client.connect(None).await) {
-            anyhow::bail!("Couldn't connect to API url");
+        loop {
+            if client.healthcheck::<HealthStatus>().await.is_ok() {
+                break;
+            }
+            async_sleep(Duration::from_secs(1)).await;
```

Comment:
Probably good to log something here at at least debug level, so we're not sleeping forever with no output